tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -684,11 +687,11 @@ def solution_vector_block(self, atomic_number, number_density):
         numpy.array
             Block of the solution vector corresponding to the current atomic number.
         """
-        solution_vector = np.zeros(atomic_number + 1)
+        solution_vector = pd.Series(0.0, index=rate_matrix_index).values
```

Comment:
Is this likely to be called many times? Because this is likely to be a performance sink.

Reply:
It's called for each shell

Reply:
what about now?